testing/file_test/file_test_base.cpp

```diff
@@ -378,8 +378,11 @@ auto FileTestBase::DoArgReplacements(
         }
         it = test_args.erase(it);
         for (const auto& file : test_files) {
-          it = test_args.insert(it, file.filename);
-          ++it;
+          const std::string& filename = file.filename;
+          if (!filename.ends_with(".h")) {
+            it = test_args.insert(it, filename);
+            ++it;
+          }
```

Comment:
Would it make sense to produce an error if we don't recognize the extension (eg, if it's not `.carbon` or `.h` for now)?

Reply:
I believe this will prevent us from testing wrong extensions errors.
[toolchain/check/testdata/packages/fail_extension.carbon](https://github.com/carbon-language/carbon-lang/blob/79ba184dabcda5003374df634a7481b639e00fdd/toolchain/check/testdata/packages/fail_extension.carbon#L18)